conans/client/graph/compatibility.py

```diff
@@ -29,7 +29,8 @@ def compatibility(conanfile):
 
 def cppstd_compat(conanfile):
     # It will try to find packages with all the cppstd versions
-
+    if not getattr(conanfile, "compatibility_cppstd", True):
```

Comment:
Lets explore the ``conf`` mechanism instead of attributes:
- With intermediate priorities, soft and hard overwritten by user profiles